usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedAgencyTest.java

```diff
@@ -0,0 +1,373 @@
+        final ValidationResultRepository mockResultRepo = mock(ValidationResultRepository.class);
+        final GtfsDataRepository mockGtfsDataRepo = mock(GtfsDataRepository.class);
+        final Agency.AgencyBuilder mockBuilder = mock(Agency.AgencyBuilder.class, RETURNS_SELF);
+        final Agency mockAgency = mock(Agency.class);
+        final ParsedEntity mockParsedAgency = mock(ParsedEntity.class);
+        @SuppressWarnings("unchecked") final List<Notice> mockNoticeCollection = mock(List.class);
+        @SuppressWarnings("rawtypes") final GenericType mockGenericType = mock(GenericType.class);
+
+        when(mockBuilder.build(mockNoticeCollection)).thenReturn(mockGenericType);
+        when(mockGenericType.getState()).thenReturn(true);
+        when(mockGenericType.getData()).thenReturn(mockAgency);
+
+        when(mockParsedAgency.get(anyString())).thenReturn(STRING_TEST_VALUE);
+
+        when(mockGtfsDataRepo.addAgency(mockAgency)).thenReturn(mockAgency);
+
+        final ProcessParsedAgency underTest = new ProcessParsedAgency(mockResultRepo, mockGtfsDataRepo, mockBuilder);
+
+        underTest.execute(mockParsedAgency, mockNoticeCollection);
+
+        final InOrder inOrder = Mockito.inOrder(mockGtfsDataRepo, mockBuilder, mockParsedAgency);
+
+        verify(mockParsedAgency, times(1)).get(ArgumentMatchers.eq(AGENCY_ID));
+        verify(mockParsedAgency, times(1)).get(ArgumentMatchers.eq(AGENCY_NAME));
+        verify(mockParsedAgency, times(1)).get(ArgumentMatchers.eq(AGENCY_URL));
+        verify(mockParsedAgency, times(1)).get(ArgumentMatchers.eq(AGENCY_TIMEZONE));
+        verify(mockParsedAgency, times(1)).get(ArgumentMatchers.eq(AGENCY_LANG));
+        verify(mockParsedAgency, times(1)).get(ArgumentMatchers.eq(AGENCY_PHONE));
+        verify(mockParsedAgency, times(1)).get(ArgumentMatchers.eq(AGENCY_FARE_URL));
+        verify(mockParsedAgency, times(1)).get(ArgumentMatchers.eq(AGENCY_EMAIL));
+
+        verify(mockBuilder, times(1)).agencyId(anyString());
+        verify(mockBuilder, times(1)).agencyName(anyString());
+        verify(mockBuilder, times(1)).agencyUrl(anyString());
+        verify(mockBuilder, times(1)).agencyTimezone(anyString());
+        verify(mockBuilder, times(1)).agencyLang(anyString());
+        verify(mockBuilder, times(1)).agencyPhone(anyString());
+        verify(mockBuilder, times(1)).agencyFareUrl(anyString());
+        verify(mockBuilder, times(1)).agencyEmail(anyString());
+
+        inOrder.verify(mockBuilder, times(1)).build(ArgumentMatchers.eq(mockNoticeCollection));
+        inOrder.verify(mockGtfsDataRepo, times(1)).addAgency(ArgumentMatchers.eq(mockAgency));
+
+        verifyNoMoreInteractions(mockBuilder, mockAgency, mockParsedAgency, mockGtfsDataRepo);
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    @Test
+    void nullAgencyNameShouldGenerateNoticeToResultRepoAndShouldNotBeAddedToGtfsDataRepo() {
```

Comment:
Test seems too specific. Should only consider behavior when mocked builder `build` method fails